src/App.tsx

```diff
@@ -1,10 +1,58 @@
 /* eslint-disable jsx-a11y/control-has-associated-label */
-import React from 'react';
+import React, { useState, useEffect, useMemo } from 'react';
 import { UserWarning } from './UserWarning';
+import { Todo } from './types/Todo';
+import { Select } from './types/Select';
+import { getTodos } from './api/todos';
+import { Main } from './components/Main';
+import { Footer } from './components/Footer';
+import { Notification } from './components/Notification';
+import { Erorrs } from './types/Erorrs';
+import { Header } from './components/Header';
 
-const USER_ID = 0;
+const USER_ID = 9960;
 
 export const App: React.FC = () => {
+  const [todoList, setTodoList] = useState<Todo[] | null>(null);
+  const [selectedFilter, setSelectedFilter] = useState(Select.all);
+  const [typeError, setTypeError] = useState<string | null>(null);
+  const [notificationError, setNotificationError] = useState(false);
+
+  const filteringList = useMemo(() => {
+    const filter = todoList && [...todoList].filter((todo) => {
```

Comment:
filter non muttable method you don't need destruct your list
```suggestion
    return todoList.filter((todo) => {
```